File: FRDIntentDemo/SecondViewController.m
//
//  SecondViewController.m
//  FRDIntent
//
//  Created by GUO Lin on 9/13/16.
//  Copyright © 2016 Douban Inc. All rights reserved.
//

#import "SecondViewController.h"

@interface SecondViewController ()

@property (nonatomic, strong) NSDictionary<NSString *, id> *data;

@end

@implementation SecondViewController

- (id)initWithExtras:(NSDictionary<NSString *,id> *)extras {
  self = [super init];
  if (self) {
    _data = extras;
  }
  return self;
}

- (void)viewDidLoad {
  [super viewDidLoad];
  self.title = @"Second View Controller";
  self.view.backgroundColor = [UIColor whiteColor];

  UILabel *numberLabel = [[UILabel alloc] init];
  numberLabel.frame = CGRectMake(20, 100, self.view.bounds.size.width - 40, 400);
  numberLabel.textAlignment = NSTextAlignmentCenter;
  numberLabel.numberOfLines = 0;
  if (self.data) {
    numberLabel.text = [NSString stringWithFormat:@"%@", self.data];
  }
  [self.view addSubview:numberLabel];
}

- (void)dealloc {
  printf("dealloc");
}

@end

File: FRDIntentDemo/SecondViewController.swift
//
//  SecondViewController.swift
//  FRDBeanFactory
//
//  Created by GUO Lin on 8/24/16.
//  Copyright © 2016 Douban Inc. All rights reserved.
//

import UIKit
import FRDIntent

class SecondViewController: UIViewController, IntentReceivable {

  var number: NSNumber?

  convenience required init(extra: Dictionary<String, AnyObject>?) {
    let number = extra?["number"] as? NSNumber
    self.init(number: number)
  }

  init(number: NSNumber?) {
    self.number = number
    super.init(nibName: nil, bundle: nil)
  }

  required init?(coder aDecoder: NSCoder) {
    fatalError("init(coder:) has not been implemented")
  }

  override func viewDidLoad() {
    super.viewDidLoad()
    title = "SecondViewController"
    view.backgroundColor = UIColor.whiteColor()
    let numberLabel = UILabel()
    numberLabel.frame = CGRect(x: 20, y: 100, width: view.bounds.size.width, height: 44)
    numberLabel.textAlignment = .Center
    numberLabel.text = "\(number!)"
    view.addSubview(numberLabel)
  }
}
